src/builtin/filter.rs

```diff
@@ -0,0 +1,58 @@
+use crate::error::Result;
+use serde_json::Value;
+
+pub type FilterFn = Box<dyn Fn(&Value) -> Result<Value>>;
+
+pub(crate) fn lower(value: &Value) -> Result<Value> {
+    let s = value.as_str().ok_or_else(|| "`lower` filter accepts string only")?;
+    Ok(Value::String(s.to_lowercase()))
+}
+
+pub(crate) fn upper(value: &Value) -> Result<Value> {
+    let s = value.as_str().ok_or_else(|| "`upper` filter accepts string only")?;
+    Ok(Value::String(s.to_uppercase()))
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
I like the level of detail in the tests here and in other places :) What do you think about adding `quickcheck` tests later ? To try finding edge cases automatically ? May be useful for a generic tool like this library.